### DISCUSSION TITLE
Disable preview button if preview returns undefined or empty string?

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#2219] [Roadmap]
Sometimes not all documents of a collection is previewable. I think it's reasonable to be disabled.

```
export type GeneratePreviewURL = (doc: Record<string, unknown>, options: GeneratePreviewURLOptions) => Promise<string | false> | string | false;
```

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/5801015/222449896-20aeb363-d41c-4c6a-bf42-b542bfb61b74.png">

**Comment by jacobsfletch** [comment]
@Stupidism I think this makes perfect sense, I'm going to mark this as `planned`.

**Comment by evelynhathaway** [comment]
Related PR that was merged after this discussion was created: #2507

If the preview function returns `null`, clicking the button shows an error toast. It's not the same as disabling or removing the button
